src/main/java/org/jenkinsci/plugins/ewm/actions/ExwsAllocateActionImpl.java

```diff
@@ -62,4 +67,28 @@ public void onAttached(Run<?, ?> run) {
     public void onLoad(Run<?, ?> run) {
         this.parent = run;
     }
+
+    @SuppressWarnings("unused")
+    public ExternalWorkspace getDynamic(String token) {
+        return getExternalWorkspaceById(token);
+    }
+
+    /**
+     * Returns the {@link ExternalWorkspace} that has the id equal to the given {@code id}.
+     * Returns {@code null} if none was found.
+     *
+     * @param id the external workspace id
+     * @return the external workspace whose id matches the given token, {@code null} otherwise
+     */
+    @CheckForNull
+    private ExternalWorkspace getExternalWorkspaceById(String id) {
```

Comment:
I have added a `synchronized` block for this in 8cb8f69.
I think it's better than having the entire method as `synchronized`.